sktime/classification/model_evaluation/_functions.py

```diff
@@ -0,0 +1,481 @@
+#!/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements functions to be used in evaluating classification models."""
+
+__author__ = ["ksharma6", "jgyasu"]
+__all__ = ["evaluate"]
+
+import inspect
+import time
+import warnings
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import KFold
+
+from sktime.datatypes import check_is_scitype, convert
+from sktime.exceptions import FitFailedWarning
+from sktime.utils.dependencies import _check_soft_dependencies
+
+PANDAS_MTYPES = ["pd.DataFrame", "pd.Series", "pd-multiindex", "pd_multiindex_hier"]
+
+
+def _is_proba_classification_score(metric) -> bool:
+    """
+    Check if metric function is intended for probabilistic classification.
+
+    This function attempts to identify if the input `metric` is a classification
+    score that expects predicted probabilities rather than class labels.
+    It performs this check using:
+
+    1. A set of known probability-based metric names.
+    2. Inspection of the function's signature for indicative argument names.
+    3. A test call using sample `y_true` and `y_pred_proba` arrays to check
+    compatibility.
+
+    Parameters
+    ----------
+    metric : callable or str
+        A metric function or its name to check.
+
+    Returns
+    -------
+    bool
+        True if the metric appears to be for probabilistic classification;
+        False otherwise.
+    """
+    PROBA_METRICS = {"brier_score_loss", "log_loss", "roc_auc_score"}
+
+    metric_name = getattr(metric, "__name__", str(metric))
+    if metric_name in PROBA_METRICS:
+        return True
+
+    try:
```

Comment:
why is a try/except needed here?

Can we try to replace it by if/else? It is best practice to avoid try/except by if/else using the exact failure condition if what we want to do is not actually error handling but type condition handling.